src/include/codegen/aggregation.h

```diff
@@ -36,17 +38,39 @@ namespace codegen {
 //===----------------------------------------------------------------------===//
 class Aggregation {
  public:
+  // Constructor taking the runtime state reference
+  Aggregation(RuntimeState &runtime_state) : runtime_state_(runtime_state) {}
+
+  // Setup the aggregation to handle the provided aggregates
+  void Setup(CompilationContext &context,
```

Comment:
I don't think you need the `CompilationContext` here. Just pass the `CodeGen` instance.